inc/class-core-sitemaps-provider.php

```diff
@@ -279,7 +279,12 @@ public function get_sitemap_lastmod( $name, $page ) {
 
 		// If blank, schedule a job.
 		if ( empty( $lastmod ) && ! wp_doing_cron() ) {
-			wp_schedule_single_event( time() + 500, 'core_sitemaps_calculate_lastmod', array( $this->slug, $name, $page ) );
+			$event_args = array( $this->slug, $name, $page );
+
+			// Don't schedule a duplicate job.
+			if ( ! wp_next_scheduled( 'core_sitemaps_calculate_lastmod', $event_args ) ) {
```

Comment:
It's good to be extra safe, although if the codex (https://codex.wordpress.org/Function_Reference/wp_schedule_single_event) is to be believed it's currently not needed:

>Note that scheduling an event to occur before 10 minutes after an existing event of the same name will be ignored, unless you pass unique values for $args to each scheduled event. See wp_next_scheduled() for more information.
>
>This behavior is subject to change, as the original intention of the code was to prevent scheduling two identical events within ten minutes of each other, not preventing scheduling of identical events until ten minutes after the next scheduled occurrence.

Reply:
`wp_schedule_single_event()` indeed takes care of making sure duplicates are not scheduled within 10 minutes +/- one that is already scheduled. In this case, I'm trying to avoid scheduling a duplicate job _ever_. Perhaps unnecessary, but that's why I've added this check.